src/app/clients/add/databaseFunctions.ts

```diff
@@ -45,6 +54,190 @@ export const insertClient = async (
     );
 };
 
-export const deleteFailedInsert = async (primary_key: string): Promise<void> => {
-    await supabase.from("clients").delete().eq("primary_key", primary_key);
+const deleteFailedInsert = async (primaryKey: string): Promise<void> => {
+    await supabase.from("clients").delete().eq("primary_key", primaryKey);
+};
+
+const updateClient = async (
+    clientRecord: ClientDatabaseUpdateRecord,
+    primaryKey: string
+): Promise<ClientDatabaseFetchRecord> => {
+    const {
+        data: ids,
+        status,
+        error,
+    } = await supabase
+        .from("clients")
+        .update(clientRecord)
+        .eq("primary_key", primaryKey)
+        .select("primary_key, family_id");
+
+    if (error === null && Math.floor(status / 100) === 2) {
+        return ids![0];
+    }
+    throw Error(
+        `Error occurred whilst updating into Clients table. HTTP Code: ${status}, PostgreSQL Code: ${error?.code}. `
+    );
+};
+
+const updateChild = async (child: Person): Promise<void> => {
+    const childRecord: FamilyDatabaseUpdateRecord = {
+        gender: child.gender,
+        age: child.age,
+    };
+    const { status, error } = await supabase
+        .from("families")
+        .update(childRecord)
+        .eq("primary_key", child.primaryKey);
+
+    if (error !== null || Math.floor(status / 100) !== 2) {
+        throw Error(
+            `Error occurred whilst updating the Families table. HTTP Code: ${status}, PostgreSQL Code: ${error?.code}. `
+        );
+    }
+};
+
+const countAdultMember = async (familyID: string, gender: string): Promise<number> => {
+    const { count, error } = await supabase
+        .from("families")
+        .select("*", { count: "exact", head: true })
+        .eq("family_id", familyID)
+        .eq("gender", gender)
+        .is("age", null);
+
+    if (error !== null || count === null) {
```

Comment:
if there are no adult members, it will return count = 0 instead of count === null (which is returned when there is an error like the table doesn't exist). The reason I have both `error !== null` and `count === null`.